File: benchmark/results/ts-invocation-inline/run-001/b-baseline/output/src/domain/ordering/services/OrderValidator.ts
import { Order } from "../aggregates/Order";

/** Service that validates Orders prior to mutation. */
export class OrderValidator {
  create(input: Partial<Order>): Order {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): Order | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Order[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}

File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/ordering/services/order_validator.py
"""OrderValidator for the ordering domain."""

from typing import Optional

from metaengine_demo.ordering.aggregates.order import Order


class OrderValidator:
    """Domain service that validates orders."""

    def create(self, input: Optional[Order]) -> Order:
        """Validate creation input for an order."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Order]:
        """Validate access for an order by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Order]:
        """Validate a list query for orders."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Validate deletion of an order by id."""
        raise NotImplementedError("not implemented")
